.github/workflows/docker-publish.yml

```diff
@@ -0,0 +1,76 @@
+name: Docker
+
+on:
+  release:
+    types: [published]
+env:
+  BUILD_VERSION: ${{ github.ref }}
+  IMAGE_NAME: cloudflared
+  DOCKER_CLI_EXPERIMENTAL: enabled
+  REPOSITORY: ${{ github.repository }}
+
+jobs:
+  # Push image to GitHub Packages.
+  # See also https://docs.docker.com/docker-hub/builds/
+  push:
+    runs-on: ubuntu-latest
+    if: ${{ github.event_name != 'pull_request' }}
+    steps:
+      - name: Source checkout
+        uses: actions/checkout@v2.3.4 
+ 
+      - name: Setup QEMU
```

Comment:
Why do we need QEMU?

Reply:
Currently it isn't doing anything, but does lay the groundwork to build multi-architecture docker images if that is wanted. https://github.com/docker/build-push-action/blob/master/docs/advanced/multi-platform.md

Reply:
I do want to note, I didn't include the multi-architecture build when I submitted this PR because I noticed the docker image is only built for amd64. If this is something we want to incorporate I can always add it.